Gems/Automation/Code/Source/AutomationSystemComponent.cpp

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AutomationSystemComponent.h>
+
+#include <AutomationScriptBindings.h>
+
+#include <AzCore/Asset/AssetCommon.h>
+#include <AzCore/Asset/AssetManager.h>
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/Script/ScriptAsset.h>
+#include <AzCore/Script/ScriptContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <AzCore/Serialization/SerializeContext.h>
+
+#include <AzFramework/API/ApplicationAPI.h>
+#include <AzFramework/Asset/AssetSystemBus.h>
+
+namespace Automation
+{
+    namespace
+    {
+        AZ::Data::Asset<AZ::ScriptAsset> LoadScriptAssetFromPath(const char* productPath)
+        {
+            char resolvedPath[AZ_MAX_PATH_LEN] = {0};
+            AZ::IO::FileIOBase::GetInstance()->ResolvePath(productPath, resolvedPath, AZ_MAX_PATH_LEN);
```

Comment:
There is an overload of FileIOBase::ResolvePath that accepts an `AZ::IO::FixedMaxPath` output parameter: https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/FileIO.h#L250